sdk/cmake/host-tools.cmake

```diff
@@ -15,10 +15,53 @@ function(setup_host_tools)
         set(HOST_EXE_SUFFIX ".exe")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH x86) # host-tools are not happy compiling for x64...
```

Comment:
This made all the bots green...

Reply:
Does that mean we will have problem when compiling for x64 architecture on x64 host? Because in this case `if(ARCH STREQUAL HOST_ARCH)` branch will be executed, where no shim is used.

Or it's somehow specifically related to only this case?

Reply:
host tools cannot be built in amd64 mode

Reply:
Sounds sad but true... :(

Reply:
> host tools cannot be built in amd64 mode
> Sounds sad but true

My custom build compiles fine with `amd64`.
What issue(s) did/do you have with it?